### DISCUSSION TITLE
ffmpeg_set_codec_preferred: Codec mpeg4 not found

### DISCUSSION THREAD

**Discussion by kristofejro** [Motion-Project/motion#1597] [General]
I have upgraded today motion from 4.4.0 to 4.5.0 and also ffmpeg was upgraded from 7:4.3.4-0+deb11u1 to 7:4.3.5-0+deb11u1

When event start (movie recording) there are messages in the log file:
```
[1:ml1] [ERR] [ENC] [Nov 02 13:08:39] ffmpeg_set_codec_preferred: Codec mpeg4 not found
[1:ml1] [ERR] [ENC] [Nov 02 13:08:39] ffmpeg_open: Failed to allocate codec!
[1:ml1] [ERR] [EVT] [Nov 02 13:08:39] event_ffmpeg_newfile: Error opening context for movie output.
```

It is probably related with new ffmpeg because after downgrading motion result is the same.
In my config I have codec set to mpeg4 (movie_codec mpeg4)

Can you give me any idea how to toubleshoot it. What and where motion is looking for and cannot find.

I am on:
Linux raspberrypi 5.15.74-v7l+ #1595 SMP Wed Oct 26 11:05:08 BST 2022 armv7l GNU/Linux
Raspbian GNU/Linux 11 (bullseye)

Thanks.

**Comment by Mr-Dave** [comment]
The 'avi' container is pretty old but if you want it, use `msmpeg4` instead.  (These are assigned in the ffmpeg.c module if you want to see)

**Reply by kristofejro** [reply]
Thank you @Mr-Dave. `msmpeg4` solves my issue.
Do you by any chance know why `mpeg4` stopped working and motion somehow cannot find the codec?
